aesara/link/numba/linker.py

```diff
@@ -0,0 +1,139 @@
+import ast
+import uuid
+
+from aesara.graph.basic import Constant
+
+from .dispatch import make_numba_thunk
+from .naming import NameFactory, unique_name_for_apply
+
+
+def create_storage_map(fgraph, order):
+    storage_map = {}
+    constants = {}
+
+    input_storage = [unique_name_for_apply(input) for input in fgraph.inputs]
+
+    for node, name in zip(fgraph.inputs, input_storage):
+        storage_map[node] = name
+
+    for node in order:
+        for r in node.inputs:
+            if r not in storage_map:
+                assert isinstance(r, Constant)
```

Comment:
Should this check for `Variable` class instance instead ?

Reply:
It should be Constant:
This check happens if we see an input to a node, that we haven't seen before, either as an input to the whole graph, or as an output of a apply instance earlier in the graph execution. The only way this should happen, is if this is a constant (or maybe a shared? Got to admit I didn't think about those at all for now...)